frontend/javascripts/dashboard/folders/metadata_table.tsx

```diff
@@ -49,22 +52,106 @@ function getMetadataTypeLabel(type: APIMetadata["type"]) {
   }
 }
 
+type EmptyMetadataPlaceholderProps = {
+  addNewEntryMenuItems: MenuProps;
+};
+const EmptyMetadataPlaceholder: React.FC<EmptyMetadataPlaceholderProps> = ({
+  addNewEntryMenuItems,
+}) => {
+  return (
+    <div className="flex-center-child empty-metadata-placeholder">
+      <img
+        src="/assets/images/metadata-teaser.svg"
+        alt="Metadata preview"
+        style={{ width: "60%", marginBottom: 16 }}
+      />
+      <span style={{ marginTop: 10 }}>
+        <Dropdown menu={addNewEntryMenuItems} placement="bottom" trigger={["click"]} autoFocus>
+          <Button icon={<PlusOutlined style={{ marginLeft: -2 }} />}>
+            Add First Metadata Entry
+          </Button>
+        </Dropdown>
+      </span>
+    </div>
+  );
+};
+
+interface MetadataValueInputProps {
+  record: APIMetadataWithError;
+  index: number;
+  focusedRow: number | null;
+  setFocusedRow: (row: number | null) => void;
+  updateMetadataValue: (index: number, newValue: number | string | string[]) => void;
+  isSaving: boolean;
+  availableStrArrayTagOptions: { value: string; label: string }[];
+}
+
+const MetadataValueInput: React.FC<MetadataValueInputProps> = ({
+  record,
+  index,
+  focusedRow,
+  setFocusedRow,
+  updateMetadataValue,
+  isSaving,
+  availableStrArrayTagOptions,
+}) => {
+  const isFocused = index === focusedRow;
+  const sharedProps = {
+    className: isFocused ? undefined : "transparent-input",
+    onFocus: () => setFocusedRow(index),
+    onBlur: () => setFocusedRow(null),
+    placeholder: "Value",
+    size: "small" as InputNumberProps<number>["size"],
+    disabled: isSaving,
+  };
+
+  switch (record.type) {
+    case APIMetadataType.NUMBER:
+      return (
+        <InputNumber
+          value={record.value as number}
+          onChange={(newNum) => updateMetadataValue(index, newNum || 0)}
+          {...sharedProps}
+        />
+      );
+    case APIMetadataType.STRING:
+      return (
+        <Input
+          value={record.value}
+          onChange={(evt) => updateMetadataValue(index, evt.target.value)}
+          {...sharedProps}
+        />
+      );
+    case APIMetadataType.STRING_ARRAY:
+      return (
+        <Select
+          mode="tags"
+          value={record.value as string[]}
+          onChange={(values) => updateMetadataValue(index, values)}
+          options={availableStrArrayTagOptions}
+          suffixIcon={null}
+          {...sharedProps}
+        />
+      );
+    default:
+      return null;
+  }
+};
+
 const saveCurrentMetadata = async (
   datasetOrFolderToUpdate: APIDataset | Folder,
   metadata: IndexedMetadataEntries,
   context: DatasetCollectionContextValue,
-  { dontUpdateState }: EffectCancelSignal = { dontUpdateState: false }, // equals whether the component is unmounted
+  // When the component is no longer mounted, state updates should not be performed.
```

Comment:
the comment can now be removed thanks to the wrapped setIsSaving, I think

Reply:
yeah right, thanks for paying attention :pray: